integration-tests/oauth-mock-test-runner/src/main/groovy/com/okta/test/mock/application/ApplicationTestRunner.groovy

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2017 Okta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.okta.test.mock.application
+
+import com.okta.test.mock.Config
+import com.okta.test.mock.Scenario
+import com.okta.test.mock.TestScenario
+import com.okta.test.mock.wiremock.HttpMock
+import groovy.text.StreamingTemplateEngine
+import org.junit.Assert
+import org.testng.annotations.AfterClass
+import org.testng.annotations.BeforeClass
+import org.testng.annotations.BeforeMethod
+import org.testng.SkipException; 
+import org.testng.util.Strings
+import org.yaml.snakeyaml.Yaml
+
+import java.lang.reflect.Method;
+import java.util.stream.Collectors
+
+import static io.restassured.RestAssured.given
+import static org.hamcrest.MatcherAssert.assertThat
+
+abstract class ApplicationTestRunner extends HttpMock {
+
+    private ApplicationUnderTest app = getApplicationUnderTest(getScenarioName())
+
+    private int mockPort
+    private int applicationPort
+    private TestScenario scenario
+
+    String getScenarioName() {
+        Scenario scenario = getClass().getAnnotation(Scenario)
+        if (scenario == null || Strings.isNullOrEmpty(scenario.value())) {
+            Assert.fail("@Scenario was not found on class '${getClass()}', you must annotate this class or override the 'getScenarioName()' method.")
+        }
+        setScenario(scenario.value())
+        return scenario.value()
+    }
+
+    int getApplicationPort() {
+        return applicationPort
+    }
+
+    int doGetMockPort() {
+        return mockPort
+    }
+
+    @BeforeMethod
+    public void checkToRun(Method method) {
+        for (String disabledTest : scenario.disabledTests) {
+            if (method.getName().equals(disabledTest)) {
+                throw new SkipException("Skipping the disabled test - " + disabledTest);
```

Comment:
We shouldn't need this, we can configure which tests are run with TestNG

Reply:
This is provide each individual libraries the ability to disable tests easily.
We wouldn't want the client libraries to configure testng xml file for each scenario.
I think this is a good convenience for people who wouldn't know testng config and want to disable tests.

Reply:
This is fine for now, but we can probably handle this better in the future.   The testng.xml is only temporary, so we can either pass in the list of scenarios, or something.
We could also scan the yaml file and only run the scenarios that are defined.

This doesn't help us skip a specific test method, but I'm sure we can do something for that as well.

Reply:
Ideally we don't want to be skipping tests, but instead not running them. Any only skip the tests that _should_ be supported. (i.e. don't include auth code flow tests for the front end libs)